### ISSUE TITLE
Add IsDefault check

### ISSUE THREAD

**Issue by dupdob** [tpierrain/NFluent#246]
Add a IsDefault check for all types. It will verify that the sut is the default value for the given type.
It has been suggested by @rhwy 

## Expected Behavior
Example:
```csharp
var x = 0;
Check.That(x).IsDefault();
object test = null;
Check.That(test).IsDefault();
```

## Context
It may be useful for some parametric tests, or may be seen as a preferred alternative to writing IsEqualTo(0) or IsNull().

**Comment by rhwy** [comment]
👍

**Comment by marcin-chwedczuk-cybervadis** [comment]
Using C# 7 default literal you can already write:
```csharp
Check.That(result).IsEqualTo(default);
```
I believe this feature can be useful only if it will provide a better failure message instead of:
```
The checked value is different from the expected one.
The checked value:
	[ResultType]
The expected value:
	[null]
```

- https://blogs.msdn.microsoft.com/mazhou/2017/06/06/c-7-series-part-3-default-literals/

**Comment by MendelMonteiro** [comment]
A side effect of this change is that all intellisense is broken when calling `IsEqualTo()` as it now takes a `object` instead of `T`.  I personally use `IsEqualTo()` a lot more often than `IsDefaultValue()`.

**Comment by dupdob** [comment]
Good point. I will check if this change was required.

**Comment by dupdob** [comment]
I have taken a look:
I am not sure why this change was introduced alongside **IsDefaultValue()**, it is definitely not required.
But, reversing it would induce a regression: a bunch of tests are no longer compiling when switching back to 'T' instead of object. And keeping both alongside leads to ambiguity related issues for some construction.

I will see if I can find an adequate design. Until then, I am afraid the signature will stay as of today.

**Comment by MendelMonteiro** [comment]
Yeah, I can see how backwards compatibility is an issue here.  It is a shame it wasn't caught earlier.

**Comment by dupdob** [comment]
Signature induced issues are the harder to catch

**Comment by dupdob** [comment]
I have implemented a hack. I need to consider wether I want to release it or not :-)

**Comment by dupdob** [comment]
Opened issue #314 to track this issue
